packages/polaris-middlewares/src/irrelevant-entities/irrelevant-entities-middleware.ts

```diff
@@ -27,25 +27,48 @@ export class IrrelevantEntitiesMiddleware {
         } as any;
     }
 
-    private static async queryIrrelevant(
-        connection: PolarisConnection,
-        tableName: string,
+    private async queryIrrelevant(
+        typeName: string,
         context: PolarisGraphQLContext,
         result: any,
     ) {
-        let irrelevantQuery = await connection
-            .getRepository(tableName)
-            .createQueryBuilderWithDeletedEntities(context, tableName)
-            .select('id');
+      if (this.connectionLessConfiguration) {
+        const irrelevantWhereCriteria: ConnectionlessIrrelevantEntitiesCriteria = {
+          realityId: context.requestHeaders.realityId || 0,
+          notInIds:
+              Array.isArray(result) && result.length > 0 ? result.map((x: any) => x.id) : [],
```

Comment:
`result.map((x: any) => x.id` happens either way, put in var before connection check